internal/impl/devbox.go

```diff
@@ -760,8 +760,15 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	}
 
 	currentEnvPath := env["PATH"]
-	if d.pure { // make nix available inside pure shell - necessary for devbox add to work
-		currentEnvPath = "/nix/var/nix/profiles/default/bin"
+	if d.pure { // make nix available inside pure shell - necessary for devbox commands to work
+		singleUserNixBin := fmt.Sprintf("%s/.nix-profile/bin", env["HOME"])
+		multiUserNixBin := "/nix/var/nix/profiles/default/bin"
+		// ~/.nix-profile/bin takes precedent because of this from Nix reference manual:
+		// "You generally wouldn’t have /nix/var/nix/profiles/some-profile/bin in your PATH.
+		// Rather, there is a symlink ~/.nix-profile that points to your current profile.
+		// This means that you should put ~/.nix-profile/bin in your PATH
+		// (and indeed, that’s what the initialisation script /nix/etc/profile.d/nix.sh does)."
+		currentEnvPath = fmt.Sprintf("%s:%s", singleUserNixBin, multiUserNixBin)
```

Comment:
From offline chat: IMO we should only keep stuff in path if it's already there and not add it if it's missing to avoid polluting the path.

Reply:
Fair enough. I updated the PR after reading more about xdg stuff and tested the PR in a fresh ubuntu install with XDG_STATE_HOME set to an arbitrary dir, and then installing nix.